pkg/controllers/management/node/cleanup.go

```diff
@@ -0,0 +1,372 @@
+		defer cancel()
+
+		_, msg, err := kubectl.Drain(ctx, kubeConfig, nodeCopy.Status.NodeName, nodehelper.GetDrainFlags(nodeCopy))
+		if ctx.Err() != nil {
+			logrus.Errorf("node [%s] kubectl drain failed, retrying: %s", nodeCopy.Spec.RequestedHostname, ctx.Err())
+			return false, nil
+		}
+		if err != nil {
+			// kubectl failed continue on with delete any way
+			logrus.Errorf("node [%s] kubectl drain error, retrying: %s", nodeCopy.Spec.RequestedHostname, err)
+			return false, nil
+		}
+
+		logrus.Infof("node [%s] kubectl drain response: %s", nodeCopy.Spec.RequestedHostname, msg)
+		return true, nil
+	})
+}
+
+func (m *Lifecycle) cleanRKENode(node *v3.Node) error {
+	cluster, err := m.clusterLister.Get("", node.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if cluster.Status.Driver != v32.ClusterDriverRKE {
+		// not an rke node, bail out
+		return nil
+	}
+
+	userContext, err := m.clusterManager.UserContext(node.Namespace)
+	if err != nil {
+		if kerror.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	job, err := m.createCleanUpJob(userContext, node)
+	if err != nil {
+		return err
+	}
+
+	return m.waitUntilJobCompletes(userContext, job)
+}
+
+func (m *Lifecycle) waitUntilJobCompletes(userContext *config.UserContext, job *v1.Job) error {
+	backoff := wait.Backoff{
+		Duration: 3 * time.Second,
+		Factor:   1,
+		Jitter:   0,
+		Steps:    10,
+	}
+
+	logrus.Infof("[node-cleanup] validating cleanup job %s finished, retrying up to 10 times", job.Name)
+	// purposefully ignoring error, if the drain fails this falls back to deleting the node as usual
+	if err := wait.ExponentialBackoff(backoff, func() (bool, error) {
+		ctx, cancel := context.WithTimeout(m.ctx, backoff.Duration)
+		defer cancel()
+
+		j, err := userContext.K8sClient.BatchV1().Jobs(job.Namespace).Get(ctx, job.Name, metav1.GetOptions{})
+		if ctx.Err() != nil {
+			logrus.Errorf("[node-cleanup] context failed while retrieving job %s, retrying: %s", job.Name, ctx.Err())
+			return false, nil
+		}
+		if err != nil {
+			// kubectl failed continue on with delete any way
+			logrus.Errorf("[node-cleanup] failed to get job %s, retrying: %s", job.Name, err)
+			return false, nil
+		}
+
+		if j.Status.Succeeded == 0 {
+			logrus.Infof("[node-cleanup] job %s still hasn't finished, backing off and retrying", job.Name)
+			return false, nil
+		}
+
+		logrus.Infof("[node-cleanup] job %s finished, continuing to delete v3 node", job.Name)
+		return true, nil
+	}); err != nil {
+		if err.Error() == "timed out waiting for the condition" {
+			return nil // gave it 30s and it timed out so move to delete v1 node
+		}
+		return err
+	}
+
+	// remove the job to clean up
+	return userContext.K8sClient.BatchV1().Jobs(job.Namespace).Delete(m.ctx, job.Name, metav1.DeleteOptions{})
+}
+
+func (m *Lifecycle) createCleanUpJob(userContext *config.UserContext, node *v3.Node) (*batchV1.Job, error) {
```

Comment:
nit: cleanup is one word.
```suggestion
func (m *Lifecycle) createCleanupJob(userContext *config.UserContext, node *v3.Node) (*batchV1.Job, error) {
```

Reply:
how does one give dagger eyes on this thing?